constructor_lab.js

```diff
@@ -0,0 +1,304 @@
+// ## Question 1
+
+// a. Write a constructor function called `Person` that has 3 properties: 
+// a first name, a last name and a middle name. Create 2 instances of a `Person`. Print one of their first names.
+
+
+function Person (firstName, lastName, middleName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.middleName = middleName;
+}
+let syn = new Person("syn", "perez", "forrest lee");
+let maria = new Person("maria", "martinez", "liza");
+
+
+// b. Write a prototype function in `Person` called `fullName` that will return a 
+// formatted string of an instance's full name. 
+// Call this method on both the instances you created in part a.
+
+Person.prototype.fullName = function() {console.log(this.firstName + " " + this.middleName + " " + this.lastName)}
+syn.fullName()
+maria.fullName()
+
+// ## Question 2
+
+// a. Create a constructor function called `Book` that has properties `title`, 
+// `author` and `rating`. Create some instances of `Book`.
+
+function Book (title, author, rating) {
+    this.title = title;
+    this.author = author;
+    this.rating = rating;
+}
+
+let theCat = new Book ("The Cat In The Hat", "Dr. Seuss", 10);
+let oneFish = new Book ("One Fish, Two Fish", "Dr. Seuss", 10);
+let fiftyShades = new Book ("50 Shades of Grey", "E.L. James", 2);
+
+// // b. Add a prototype function to `Book` called `isGood` that returns 
+// `true` if its rating is greater than or equal to 7
+
+Book.prototype.isGood = function () {this.rating > 7 ? console.log(true) : console.log(false)}
+theCat.isGood();
+fiftyShades.isGood();
+
+// ## Question 3
+
+// // a. Create a `Dog` constructor function with four properties: 
+// `name (string), breed (string), mood (string), and hungry (boolean)`.
+
+function Dog (name, breed, mood, hungry) {
+    this.name = name;
+    this.breed = breed;
+    this.mood = mood;
+    this.hungry = hungry;
+}
+let dmx = new Dog ("DMX", "Ruff Riders", "Angry", false);
+let snoop = new Dog ("Snoop", "Doggy Dog", "Chill", true)
+
+// // b. Add a prototype function called `playFetch`. It should set the dog's 
+// `hungry` property to `true`, set its mood property to `playful`, and log "Ruff!"
+
+Dog.prototype.playFetch = function() {
+    if (this.hungry === false) {
```

Comment:
another way to write this would be.

`if(!this.hungry)`

also keep in mind that `playFetch()` would not work if the dog not hungry and it would not execute.